internal/api/rolebindings.go

```diff
@@ -0,0 +1,366 @@
+package api
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/labstack/echo/v4"
+	"go.infratographer.com/x/gidx"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/trace"
+
+	"go.infratographer.com/permissions-api/internal/iapl"
+	"go.infratographer.com/permissions-api/internal/types"
+)
+
+func resourceToSubject(subjects []types.RoleBindingSubject) []roleBindingSubject {
+	resp := make([]roleBindingSubject, len(subjects))
+	for i, subj := range subjects {
+		resp[i] = roleBindingSubject{
+			ID:        subj.SubjectResource.ID,
+			Type:      subj.SubjectResource.Type,
+			Condition: nil,
+		}
+	}
+
+	return resp
+}
+
+func (r *Router) roleBindingCreate(c echo.Context) error {
+	resourceIDStr := c.Param("id")
+
+	ctx, span := tracer.Start(
+		c.Request().Context(), "api.roleBindingCreate",
+		trace.WithAttributes(attribute.String("id", resourceIDStr)),
+	)
+	defer span.End()
+
+	resourceID, err := gidx.Parse(resourceIDStr)
+	if err != nil {
+		return r.errorResponse("error parsing resource ID", fmt.Errorf("%w: %s", ErrInvalidID, err.Error()))
+	}
+
+	var body roleBindingRequest
+
+	err = c.Bind(&body)
+	if err != nil {
+		return r.errorResponse(err.Error(), ErrParsingRequestBody)
+	}
+
+	resource, err := r.engine.NewResourceFromID(resourceID)
+	if err != nil {
+		return r.errorResponse("error creating resource", err)
+	}
+
+	subjectResource, err := r.currentSubject(c)
+	if err != nil {
+		return err
+	}
+
+	// permissions on role binding actions, similar to roles v1, are granted on the resources
+	if err := r.checkActionWithResponse(ctx, subjectResource, string(iapl.RoleBindingActionCreate), resource); err != nil {
+		return err
+	}
+
+	roleID, err := gidx.Parse(body.RoleID)
+	if err != nil {
+		return r.errorResponse("error parsing role ID", fmt.Errorf("%w: %s", ErrInvalidID, err.Error()))
+	}
+
+	roleResource, err := r.engine.NewResourceFromID(roleID)
+	if err != nil {
+		return r.errorResponse("error creating role resource", err)
+	}
+
+	subjects := make([]types.RoleBindingSubject, len(body.Subjects))
+
+	for i, s := range body.Subjects {
+		subj, err := r.engine.NewResourceFromID(s.ID)
+		if err != nil {
+			return r.errorResponse("error creating subject resource", err)
+		}
+
+		subjects[i] = types.RoleBindingSubject{
+			SubjectResource: subj,
+			Condition:       nil,
+		}
+	}
+
+	rb, err := r.engine.CreateRoleBinding(ctx, resource, roleResource, subjects)
+	if err != nil {
+		return r.errorResponse("error creating role-binding", err)
+	}
+
+	return c.JSON(
+		http.StatusOK,
+		roleBindingResponse{
+			ID:       rb.ID,
+			Subjects: resourceToSubject(rb.Subjects),
+			Role: roleBindingResponseRole{
+				ID:   rb.Role.ID,
+				Name: rb.Role.Name,
+			},
+		},
+	)
+}
+
+func (r *Router) roleBindingsList(c echo.Context) error {
+	resourceIDStr := c.Param("id")
+	roleIDStr := c.QueryParam("role_id")
```

Comment:
Does enabling filtering by a specific role ID on the resource give us anything yet? Seems like a feature we can add in later.